dbt/include/trino/macros/materializations/table.sql

```diff
@@ -8,51 +9,58 @@
                                                 schema=schema,
                                                 database=database,
                                                 type='table') -%}
-  {%- set intermediate_relation = api.Relation.create(identifier=tmp_identifier,
-                                                      schema=schema,
-                                                      database=database,
-                                                      type='table') -%}
 
-  {%- set backup_relation_type = 'table' if old_relation is none else old_relation.type -%}
-  {%- set backup_relation = api.Relation.create(identifier=backup_identifier,
-                                                schema=schema,
-                                                database=database,
-                                                type=backup_relation_type) -%}
-
-  {%- set exists_as_table = (old_relation is not none and old_relation.is_table) -%}
-  {%- set exists_as_view = (old_relation is not none and old_relation.is_view) -%}
+  {% if table_exists_strategy == 'RENAME' %}
```

Comment:
Check whether the constant used for determining the handling for table existence is valid.
Use https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/incremental/on_schema_change.sql as template.

Reply:
Please see https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/incremental/on_schema_change.sql#L3